src/net/metaplaylist/isobmff_patcher.ts

```diff
@@ -0,0 +1,461 @@
+            break;
+          case "tfhd":
+            output = this.tfhd(data);
+            break;
+          case "mfhd":
+            output = this.mfhd(data);
+            break;
+          case "trun":
+            output = this.trun(data);
+            break;
+          case "sidx":
+            output = this.sidx(data, false);
+            break;
+          case "tfdt":
+            output = this.tfdt(data);
+            break;
+          case "mdat":
+            output = data;
+            break;
+          case "mvhd":
+            output = this.mvhd(data);
+            break;
+          case "tkhd":
+            output = this.tkhd(data);
+            break;
+          default:
+            output = data;
+            break;
+        }
+        // if (output.toString() !== data.toString()) {
+        //   console.log(boxtype);
+        //   console.log("");
+        // }
+      }
+      return output;
+    }
+
+    tkhd(input: Uint8Array): Uint8Array {
+      const version = input[8];
+      let output = new Uint8Array(0);
+      if (version === 1) {
+        output = concat(
+          input.subarray(0,36),
+          itobe8(0),
+          input.subarray(44, input.length)
+        );
+      } else {
+        output = concat(
+          input.subarray(0,28),
+          itobe4(0),
+          input.subarray(32, input.length)
+        );
+      }
+      return output;
+    }
+
+    mvhd(input: Uint8Array): Uint8Array {
+      const version = input[8];
+      let output = new Uint8Array(0);
+      if (version === 1) {
+        output = concat(
+          input.subarray(0,32),
+          itobe8(0),
+          input.subarray(40, input.length)
+        );
+      } else {
+        output = concat(
+          input.subarray(0,24),
+          itobe4(0),
+          input.subarray(28, input.length)
+        );
+      }
+      return output;
+    }
+
+    /**
+     * Process styp and make sure lmsg presence follows the lmsg flag parameter.
+     * Add scte35 box if appropriate
+     * @param {Uint8Array} input
+     * @return {Uint8Array} patched box
+     */
+    styp(input: Uint8Array): Uint8Array {
+      const lmsg = this.lmsg;
+      const size = be4toi(input, 0);
+      let pos = 8;
+      const brands = [];
+      while (pos < size) {
+          const brand = input.subarray(pos, pos + 4);
+          if (bytesToStr(brand) !== "lmsg") {
+              brands.push(brand);
+          }
+          pos += 4;
+      }
+      if (lmsg) {
+        brands.push(strToBytes("lmsg"));
+      }
+      const dataSize = brands.length * 4;
+      const data = new Uint8Array(dataSize);
+      for (let i = 0; i < brands.length; i++) {
+        data.set(brands[i], i*4);
+      }
+      return Atom("styp", data);
+    }
+
+    /**
+     * Process tfhd (assuming that we know the ttml size size).
+     * @param {Uint8Array} input
+     * @return {Uint8Array} patched box
+     */
+    tfhd(input: Uint8Array): Uint8Array {
+      let output = new Uint8Array(0);
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Remove this useless assignment to local variable "output". ([typescript:S1854](https://sonarcloud.io/organizations/rx-player/rules#rule_key=typescript%3AS1854))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=rx-player&pullRequest=178&issues=AWNyk0oJTmDVX2bfxyaw&open=AWNyk0oJTmDVX2bfxyaw)

[ISSUE_KEY]: AWNyk0oJTmDVX2bfxyaw